tf2_geometry_msgs/CMakeLists.txt

```diff
@@ -42,27 +48,20 @@ if(BUILD_TESTING)
 
   ament_add_gtest(test_tf2_geometry_msgs test/test_tf2_geometry_msgs.cpp)
   if(TARGET test_tf2_geometry_msgs)
-    target_include_directories(test_tf2_geometry_msgs PUBLIC include)
-    ament_target_dependencies(test_tf2_geometry_msgs
-      "geometry_msgs"
-      "orocos_kdl"
-      "rclcpp"
-      "tf2"
-      "tf2_ros"
-    )
+    target_link_libraries(test_tf2_geometry_msgs
+      ${PROJECT_NAME}
+      rclcpp::rclcpp)
```

Comment:
I feel like we should still have an explicit dependency on `tf2_ros` here, as the test directly depends upon it.

Reply:
I chose to not to make the test `target_link_libraries(... tf2_ros::tf2_ros)` to make sure the `${PROJECT_NAME}` target is properly exporting it. Thoughts?

Reply:
Summarizing our off-github chat - I'll update these PRs to comment which dependencies are being gotten from other targets in the same CMake package. This retains a check on if the exported targets are properly exporting their dependencies, while also allowing future readers of the code to know what's being depended upon and why it wasn't in the CMake file.

Reply:
Commented that `tf2_ros::tf2_ros` is required but intentionally not linked in 983cd0d